### ISSUE TITLE
New debugger does not attach to process

### ISSUE THREAD

**Issue by yuatpocketgems** [microsoft/vscode-js-debug#623]
**Describe the bug**
Previously we were able to launch a process and have inspector attach to 9229 port and use break points.

On node12 + VSCode 1.47.1, the same setup broke.
1. Our node process is started with --inspect-brk=0.0.0.0 and in console, the process is stuck at `Debugger listening on ws://0.0.0.0:9229/hash`
2. Auto-attach does not attach to 0.0.0.0:9229
3. Manually attaching via cmd palette fails with connection refused
4. With an attach configuration, we are able to attach to the process, but break points don't work.
5. Adding "debug.javascript.usePreview": false does unblock us in the meantime: attaching inspector & breakpointing.

**To Reproduce**
1. Use node12 + VSCode 1.47 (no babel etc)
2. Launch a js process
3. Debugger fails to attach.

**Log File**
I'm emailing a log file to the email address with this issue's link as reference.

**VS Code Version:** 1.47.1

**Additional context**
We do have a hierarchy of indirections before invoking nodemon which might be affecting auto-attach feature.
    * In code-workspace, we have launch via npm config
    * In node package, we have a script called `debug-vscode` which calls a bash script
    * In the bash script, we start a docker instance 
    * Then calls `docker-compose exec nodemon --inspect-brk=0.0.0.0 ...`

**Comment by connor4312** [comment]
For auto attach: https://github.com/microsoft/vscode/issues/102057#issuecomment-657728467

> With an attach configuration, we are able to attach to the process, but break points don't work.

Can you share a trace log using the instructions from the issue template?

**Comment by yuatpocketgems** [comment]
Two logs, one with autoattach and one without, were sent to connor@xbox.com.

**Comment by connor4312** [comment]
Thanks for the logs and context. It seems like you're in the same boat as https://github.com/microsoft/vscode-js-debug/issues/586#issuecomment-658958272 -- see that comment for resolution (the nightly build referenced should be available in about 2hrs).

If you need to debug until then you can set `debug.javascript.usePreview: false` to use the old debugger -- but if things work, please continue using the new one 🙂  (and if they don't, let me know!)

**Comment by yuatpocketgems** [comment]
With `debug.javascript.usePreview: false` the debugger worked for 1.47.1 and 1.47.2.

It stopped working again on 1.47.3. I believe this is a new issue, and the old issue is not resolved either.

I tried toggle debug.javascript.usePreview to `true` just now, and the process failed to attach. With the flag on, debugger says it's attached, but breakpoints are not working.

Let me know if you would like more log files.

**Comment by connor4312** [comment]
Thanks for the followup. Can you send a new trace log over?

**Comment by yuatpocketgems** [comment]
See attached logs.

**Comment by yuatpocketgems** [comment]
Just cleaned up docker & restarted vscode. This log file would probably be
cleaner than the previous.

**Comment by connor4312** [comment]
Attaching logs to mails to Github unfortunately don't come through. Please upload the log by dragging it onto the comment box on github.com, or cc me at connor@xbox.com. Thanks!

**Comment by connor4312** [comment]
Up a few weeks ago I linked a solution that involved changing the `port: 9229` property to `attachSimplePort: 9229`. Did that solution not work for you? In the logs you sent I see that it looks like you're using the plain `port` property.

**Comment by yuatpocketgems** [comment]
Tried both `attachSimplePort:0` and `attachSimplePort:9229`. Neither worked. Sending you additional logs.

**Comment by connor4312** [comment]
Ah, apologies, missing a step -- the attachSimplePort option was added in the [nightly build](https://github.com/microsoft/vscode-js-debug#nightly-extension), so you will need to be on that (or wait until the next VS Code stable release tomorrow or Thursday)

**Comment by yuatpocketgems** [comment]
Alright. I tried a couple of configurations and below are some findings.

After switching to nightly build v2020.8.1017, and turning on preview
  - With `nodemon --inspect-brk=0.0.0.0`
       - a. With `attachSimplePort: 9229`
             The debugger is able to attach in the initial node execution, then fails at the subsequent executions.
       - b. With `attachSimplePort: 0`
              The debugger fails to attach.

     It seems like the `attachSimplePort` flag doesn't work well with custom addresses?

  - With `nodemon` (without `--inspect-brk` at all)
       - c. With `attachSimplePort: 0` or `attachSimplePort: 9229`
               Debugger attached ok, but breakpoints failed to work.


Will send logs for `c` to the email thread.

**Comment by connor4312** [comment]
What's the actual command you're running in `debug-vscode`?

**Comment by yuatpocketgems** [comment]
In `package.json`

```
  "scripts": {
    ...
    "debug-vscode": "../../../scripts/service-commands.sh `pwd` debug-vscode",
  }
```

In `service-commands.sh`

  ```
  docker-compose exec $containerID \
    nodemon --no-lazy --legacy-watch --inspect-brk=0.0.0.0 --watch ... \
    jest --config=... --runInBand
```

**Comment by connor4312** [comment]
Okay. So you definitely want `attachSimplePort: 9229` then.

> then fails at the subsequent executions.

How does it fail?

**Comment by yuatpocketgems** [comment]
When nodemon sees a change in file, it re-runs jest, then the console gets stuck at 

```
Debugger listening on ws://0.0.0.0:9229/hash
For help, see: https://nodejs.org/en/docs/inspector
```

**Comment by connor4312** [comment]
js-debug doesn't support re-attaching when launching a program (without restarting the program). For your scenario I suggest moving your docker launch into a prelaunch task, and then use an "attach" config, e.g.

**tasks.json**

```json
{
  "version": "2.0.0",
  "tasks": [
    {
      "type": "npm",
      "script": "debug-vscode",
      "problemMatcher": [],
      "isBackground": true,
      "label": "npm: debug-vscode",
    }
  ]
}

```

**launch.json**

```json
{
  "version": "0.2.0",
  "configurations": [
      {
        "type": "node",
        "request": "attach",
        "name": "Attach",
        "port": 9229,
        "restart": true, // re-attach on reboot
        "preLaunchTask": "npm: debug-vscode", // docker run beforehand
      }
  ]
}
```

```


Or, use our [remote extensions](https://code.visualstudio.com/docs/remote/containers) to work right in the contain.
